src/event.rs

```diff
@@ -67,6 +67,16 @@ pub enum Event<'a, T: 'static> {
         event: DeviceEvent,
     },
 
+    /// Emitted on macOS to let the application know that a file or files are requested to be opened.
+    ///
+    /// This for example happens when the user double-clicks on a file in Finder and the file's type
+    /// is associated with this application. More precisely this event is emitted when either of
+    /// `application:openFile:` or `application:openFiles:` is called on the application delegate.
+    ///
+    /// Note that to allow associating file types with an application, said application must have
+    /// an appropriate `Info.plist` file in the application bundle defining `CFBundleDocumentTypes`.
+    OpenFiles(Vec<PathBuf>),
```

Comment:
I just noticed that this is not a device or window event. That's probably going to cause significant headache downstream. Though I'm not entirely sure what to do about it. I don't think it makes sense to attach this to any device or window.

Reply:
Just out of curiosity, why does this cause headache downstream? The only thing I can think of is having to add in a match arm about once per application. Why is that a cause of significant headache?

Reply:
All events are either WindowEvents or DeviceEvents. This would be the first event that doesn't come from winit that is a "root event".